### PR TITLE
Remove legacy behavior for link blocks

### PR DESCRIPTION
**Pull request by johnnyomair**
Since Next 13 the `Link` component no longer requires a child `<a>` tag by default. This new behavior is now used by our `InternalLinkBlock`, which uses `Link` internally. To ensure that all other link blocks (e.g., `ExternalLinkBlock`) can be styled in the application, we now render an `<a>` tag for all those blocks.

This change adds the `className` prop to the `LinkBlock` to allow styling the block. All link block usages have been adapted to directly style the block.

---

COM-936

### CODE REVIEW

File: site/src/common/components/Button.tsx [hunk 1]
```diff
@@ -1,26 +1,40 @@
...
+                pointer-events: none;
+                ${disabledButtonVariantStyle[variant]};
+            }
+
+            /* <a> doesn't support :disabled selector */
+            ${disabled &&
+            css`
+                pointer-events: none;
+                ${disabledButtonVariantStyle[variant]};
+            `}
```
Comment (johnnyomair) [inline site/src/common/components/Button.tsx:35]: We could discuss removing this altogether. Disabled links don't make any sense to me... In the rich text block we don't render the link, and in link lists we hide the link.

Comment (johnnyomair) [reply site/src/common/components/Button.tsx:35]: Also, this isn't enough for accessibility. We would need something like `<a role="link" aria-disabled="true">Learn something!</a>`

But this isn't recommended:

> It is NOT RECOMMENDED to use aria-disabled="true" on an a element with an href attribute.

https://w3c.github.io/html-aria/#el-a

But it works with `role="link"` 😂 

> Where it is [not recommended](https://w3c.github.io/html-aria/#el-a) to use the aria-disabled attribute on an <a href> element, it’s perfectly fine to use this attribute on a role=link.

https://www.scottohara.me/blog/2021/05/28/disabled-links.html

In conclusion, I'd still remove it 😁

Comment (jamesricky) [reply site/src/common/components/Button.tsx:35]: I agree, probably makes sense to remove it 😄

File: site/src/layout/header/Header.tsx [hunk 2]
```diff
@@ -68,13 +72,17 @@ const TopLevelLinkContainer = styled.li`
...
     text-decoration: none;
     padding: 5px 10px;
 
     &:hover {
         text-decoration: underline;
     }
+
+    &.active {
+        color: ${({ theme }) => theme.palette.primary.main};
+    }
```
Comment (jamesricky) [inline site/src/layout/header/Header.tsx:85]: The `active` class-name could easily be confused with the `a:active` selector. 
Maybe change it to something else... `selected`?

**Review by nsams — APPROVED** [review]

**Review by SebiVPS — APPROVED** [review]

File: site/src/layout/header/Header.tsx (hunk 2 above)
Comment (johnnyomair) [reply site/src/layout/header/Header.tsx:85]: This was inspired by React Router, which also uses active: https://v5.reactrouter.com/web/api/NavLink/activeclassname-string

I believe "active" is the term that most devs use for the currently active page, should we really change that?

File: site/src/common/components/Button.tsx (hunk 1 above)
Comment (johnnyomair) [reply site/src/common/components/Button.tsx]: Removed: c1d8bdf9ee123c0d660a8559338aa5607ecfb2b2

File: site/src/layout/header/Header.tsx (hunk 2 above)
Comment (jamesricky) [reply site/src/layout/header/Header.tsx:85]: For me at least, "active" on `<a />` means "currently being clicked", not "page I'm currently on". 
But if react router also uses that name it's probably fine.

**Review by jamesricky — APPROVED** [review]

**Review by thomasdax98 — APPROVED** [review]

**Review by mariokemetinger — APPROVED** [review]

**Pull request merged**